liblttng-ust-java-agent/java/lttng-ust-agent-jul/org/lttng/ust/agent/jul/LttngLogHandler.java

```diff
@@ -134,13 +156,17 @@ public void publish(LogRecord record) {
 		 */
 		LttngJulApi.tracepointWithContext(formattedMessage,
 				record.getLoggerName(),
-				record.getSourceClassName(),
-				record.getSourceMethodName(),
+				logSource ? record.getSourceClassName() : EMPTY,
+				logSource ? record.getSourceMethodName() : EMPTY,
 				record.getMillis(),
 				record.getLevel().intValue(),
 				record.getThreadID(),
 				contextInfo.getEntriesArray(),
 				contextInfo.getStringsArray());
 	}
 
+	public void setLogSource(boolean doLog) {
```

Comment:
Ok fair enough, if we look at Handler.class it does have several setters for these config options that can also be set via the config file. We can stay consistent with that. You could also add a getter in this case ;)